cant seem to use msgpack serializer when using sparse run

The test for the serialiser is in submit https://github.com/Parsely/streamparse/blob/master/streamparse/cli/submit.py#L78-90

but not obviously in run, so setting `topology.multilang.serializer=com.yelp.pyleus.serializer.MessagePackSerializer` in sparse run breaks with 
```
[Thread-16-some-bolt-executor[3 3]] ERROR o.a.s.d.executor - 
java.lang.RuntimeException: Error when launching multilang subprocess
Traceback (most recent call last):
  File "...venv/bin/streamparse_run", line 11, in <module>
    sys.exit(main())
  File "...venv/local/lib/python2.7/site-packages/streamparse/run.py", line 37, in main
    cls(serializer=args.serializer).run()
  File "...venv/local/lib/python2.7/site-packages/pystorm/component.py", line 476, in run
    storm_conf, context = self.read_handshake()
  File "...venv/local/lib/python2.7/site-packages/pystorm/component.py", line 303, in read_handshake
    msg = self.read_message()
  File "...venv/local/lib/python2.7/site-packages/pystorm/component.py", line 279, in read_message
    return self.serializer.read_message()
  File "...venv/local/lib/python2.7/site-packages/pystorm/serializers/json_serializer.py", line 70, in read_message
    line = self.input_stream.readline()
  File "...venv/lib/python2.7/codecs.py", line 314, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf8' codec can't decode byte 0x83 in position 0: invalid start byte

	at org.apache.storm.utils.ShellProcess.launch(ShellProcess.java:89) ~[storm-core-1.0.2.jar:1.0.2]
	at org.apache.storm.task.ShellBolt.prepare(ShellBolt.java:131) ~[storm-core-1.0.2.jar:1.0.2]
	at org.apache.storm.daemon.executor$fn__8058$fn__8071.invoke(executor.clj:795) ~[storm-core-1.0.2.jar:1.0.2]
	at org.apache.storm.util$async_loop$fn__624.invoke(util.clj:482) [storm-core-1.0.2.jar:1.0.2]
	at clojure.lang.AFn.run(AFn.java:22) [clojure-1.7.0.jar:?]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_101]
Caused by: java.io.EOFException
	at org.msgpack.io.StreamInput.readByte(StreamInput.java:60) ~[some_project-0.0.1-SNAPSHOT-standalone.jar:?]
	at org.msgpack.unpacker.MessagePackUnpacker.getHeadByte(MessagePackUnpacker.java:66) ~[some_project-0.0.1-SNAPSHOT-standalone.jar:?]
	at org.msgpack.unpacker.MessagePackUnpacker.trySkipNil(MessagePackUnpacker.java:396) ~[some_project-0.0.1-SNAPSHOT-standalone.jar:?]
	at org.msgpack.template.MapTemplate.read(MapTemplate.java:59) ~[some_project-0.0.1-SNAPSHOT-standalone.jar:?]
	at org.msgpack.template.MapTemplate.read(MapTemplate.java:27) ~[some_project-0.0.1-SNAPSHOT-standalone.jar:?]
	at org.msgpack.template.AbstractTemplate.read(AbstractTemplate.java:31) ~[some_project-0.0.1-SNAPSHOT-standalone.jar:?]
	at org.msgpack.MessagePack.read(MessagePack.java:527) ~[some_project-0.0.1-SNAPSHOT-standalone.jar:?]
	at org.msgpack.MessagePack.read(MessagePack.java:496) ~[some_project-0.0.1-SNAPSHOT-standalone.jar:?]
	at com.yelp.pyleus.serializer.MessagePackSerializer.readMessage(MessagePackSerializer.java:205) ~[some_project-0.0.1-SNAPSHOT-standalone.jar:?]
	at com.yelp.pyleus.serializer.MessagePackSerializer.connect(MessagePackSerializer.java:67) ~[some_project-0.0.1-SNAPSHOT-standalone.jar:?]
	at org.apache.storm.utils.ShellProcess.launch(ShellProcess.java:85) ~[storm-core-1.0.2.jar:1.0.2]
``` 

I think the second bolt in my topologie is trying to interpret a msgpack message with a json deserializers.

any way to get the correct setting on the shell bolt so that it uses msgpack? I was hoping to do this as i have a downstream cassandra database writer bolt and they expect uuids as 16 bytes (128 bits) and msgpack seems to be able to handle this (as does the storm native kryo) but json can't encode byte arrays like this. And its a real pain to have to submit the topologie every time instead of getting the debug out from run.

thanks

--- a/streamparse/cli/run.py
+++ b/streamparse/cli/run.py
@@ -14,9 +14,9 @@ from tempfile import NamedTemporaryFile
 from fabric.api import local, show
 from ruamel import yaml
 
-from ..util import (get_env_config, get_topology_definition,
+from ..util import (get_config, get_env_config, get_topology_definition,
                     get_topology_from_file, local_storm_version,
-                    storm_lib_version)
+                    set_topology_serializer, storm_lib_version)
 from .common import (add_ackers, add_debug, add_environment, add_name,
                      add_options, add_workers, resolve_options)
 from .jar import jar_for_deploy
@@ -25,8 +25,11 @@ from .jar import jar_for_deploy
 def run_local_topology(name=None, env_name=None, time=0, options=None):
     """Run a topology locally using Flux and `storm jar`."""
     name, topology_file = get_topology_definition(name)
+    config = get_config()
     env_name, env_config = get_env_config(env_name)
     topology_class = get_topology_from_file(topology_file)
+
+    set_topology_serializer(env_config, config, topology_class)
 
     storm_options = resolve_options(options, env_config, topology_class, name)
     if storm_options['topology.acker.executors'] != 0:

--- a/streamparse/cli/submit.py
+++ b/streamparse/cli/submit.py
@@ -19,7 +19,8 @@ from ..thrift import ShellComponent
 
 from ..util import (activate_env, get_config, get_env_config,
                     get_nimbus_client, get_topology_definition,
-                    get_topology_from_file, ssh_tunnel, warn)
+                    get_topology_from_file, set_topology_serializer,
+                    ssh_tunnel, warn)
 from .common import (add_ackers, add_debug, add_environment, add_name,
                      add_options, add_override_name, add_requirements,
                      add_timeout, add_wait, add_workers, resolve_options)
@@ -76,20 +77,7 @@ def _submit_topology(topology_name, topology_class, remote_jar_path, config,
         print("Routing Python logging to {}.".format(options['pystorm.log.path']))
         sys.stdout.flush()
 
-    serializer = env_config.get('serializer', config.get('serializer', None))
-    if serializer is not None:
-        # Set serializer arg in bolts
-        for thrift_bolt in itervalues(topology_class.thrift_bolts):
-            inner_shell = thrift_bolt.bolt_object.shell
-            if inner_shell is not None:
-                inner_shell.script = '-s {} {}'.format(serializer,
-                                                       inner_shell.script)
-        # Set serializer arg in spouts
-        for thrift_spout in itervalues(topology_class.thrift_spouts):
-            inner_shell = thrift_spout.spout_object.shell
-            if inner_shell is not None:
-                inner_shell.script = '-s {} {}'.format(serializer,
-                                                       inner_shell.script)
+    set_topology_serializer(env_config, config, topology_class)
 
     print("Submitting {} topology to nimbus...".format(topology_name), end='')
     sys.stdout.flush()

--- a/streamparse/util.py
+++ b/streamparse/util.py
@@ -20,7 +20,7 @@ from fabric.api import env, hide, local, settings
 from fabric.colors import red, yellow
 from pkg_resources import parse_version
 from prettytable import PrettyTable
-from six import iteritems
+from six import iteritems, itervalues
 from six.moves.socketserver import UDPServer, TCPServer
 from thriftpy.protocol import TBinaryProtocolFactory
 from thriftpy.rpc import make_client
@@ -444,3 +444,26 @@ def get_topology_from_file(topology_file):
     else:
         raise ValueError('Could not find topology subclass in topology module.')
     return topology_class
+
+
+def set_topology_serializer(env_config, config, topology_class):
+    """Go through the components in a `Topology` and set the serializer.
+
+    This is necessary because the `Topology` class has no information about the
+    user-specified serializer, but it needs to be passed as an argument to
+    ``streamparse_run``.
+    """
+    serializer = env_config.get('serializer', config.get('serializer', None))
+    if serializer is not None:
+        # Set serializer arg in bolts
+        for thrift_bolt in itervalues(topology_class.thrift_bolts):
+            inner_shell = thrift_bolt.bolt_object.shell
+            if inner_shell is not None:
+                inner_shell.script = '-s {} {}'.format(serializer,
+                                                       inner_shell.script)
+        # Set serializer arg in spouts
+        for thrift_spout in itervalues(topology_class.thrift_spouts):
+            inner_shell = thrift_spout.spout_object.shell
+            if inner_shell is not None:
+                inner_shell.script = '-s {} {}'.format(serializer,
+                                                       inner_shell.script)

